plugin/audit_log/audit_log.c

```diff
@@ -785,6 +796,39 @@ int is_event_class_allowed_by_policy(unsigned int class,
   return (class_mask[policy] & (1 << class)) != 0;
 }
 
+static const CHARSET_INFO *charset_info= &my_charset_latin1;
+
+static
+const char *next_word(const char *str, size_t *len)
+{
+  while (*str && my_isspace(charset_info, *str))
+  {
+    if (*str == '/' && str[1] == '*' && str[2] == '!')
+      str+= 3;
+    else if (*str == '/' && str[1] == '*')
+    {
+      while (*str && !(*str == '*' && str[1] == '/'))
+        str++;
+    }
+    else
+      str++;
+  }
+
+  *len= 0;
+  while (str[*len] && my_isvar(charset_info, str[*len]))
```

Comment:
I found what was wrong! `mtr` communicating with server in `latin1` by default. It was the source of these funny issues.
When I `set names utf8` I can `use ąžąžąžą` without backticks. And it was an issue for plugin. Solution is to use `event_general->general_charset` when parsing the query.
Database name in the connection event comes in `utf8mb3`.
So, there is another bug! Query is written to log in client charset while other fields in `utf8`.